### ISSUE TITLE
Taxon Issues

### ISSUE THREAD

**Issue by hattrill** [geneontology/go-ontology#18657]
I have heart and muscle terms that are giving taxon errors in the GO QC check pipeline.
All of the cardiac ones are under "heart development" GO:0007507 which has the narrow synonym: dorsal vessel development - so should be valid for flies.

GO:0003143	embryonic heart tube morphogenesis
GO:0003144	embryonic heart tube formation
GO:0003151	outflow tract morphogenesis
GO:0003007	heart morphogenesis
GO:0003344	pericardium morphogenesis
GO:0003319	cardioblast migration to the midline involved in heart rudiment formation
GO:0055007	cardiac muscle cell differentiation
GO:0055013	cardiac muscle cell development
GO:0060914	heart formation
GO:0035050	embryonic heart tube development
GO:0061343	cell adhesion involved in heart morphogenesis
GO:0048738	cardiac muscle tissue development

These muscle terms should be ok for flies as well:
GO:0007519	skeletal muscle tissue development
GO:0016203	muscle attachment
GO:0016204	determination of muscle attachment site

Not sure where the constraint is coming from - probably Uberon, but can't seem to pin it down. 

As a side note, Protein2GO isn't seeing these constraints - not sure if this is a synchrony issue -  I usually rely on the P2GO report for picking up taxon violations, so I haven't looked at the GO taxon errors for a while (the renal ones were picked up by P2GO).

**Comment by ukemi** [comment]
skeletal muscle?

**Comment by ukemi** [comment]
The UBERON references to heart need to be changed for terms that refer to the relaxed taxonomic structure. Note that heart_development refers to primary circulatory organ.

**Comment by hattrill** [comment]
Doesn't have to be attached to a bone skeleton....it's somatic muscle. It's in the Fly Anatomy cv e.g. skeletal muscle of abdomen FBbt:00003447

**Comment by ukemi** [comment]
Hmm. The GO term refers to skeletal muscle tissue in UBERON (0001134). This is separate from somatic muscle in UBERON (0014895). We have somatic muscle development in GO that refers to the invertebrate term. I suggest we keep in synch with UBERON and leave these taxon restrictions in place, using the somatic muscle term for flies and retaining the skeletal muscle tissue/organ hierarchy as is.

**Comment by hattrill** [comment]
ok - I can live with that. 

wrt 'muscle attachment'  - can we make that less skeletal? The definition is "The developmental process in which a skeletal muscle attaches to its target (such as bone or body wall)." - removing 
"skeletal" and change it to part of "muscle structure development"

In FBbt: muscle attachment site  "The place where the muscle system attaches to the body wall"

**Comment by ukemi** [comment]
I'm going to fix everything I can in GO, but I think some of the more specific structures will need to be modified in UBERON. For example, the paper that you use for 'cardioblast migration to the midline involved in heart rudiment formation' specifically talks about the similarities in flies and vertebrates. These are the kinds of things we want to capture in GO and is why I worry sometimes about taxon constraints being too stringent.

**Comment by hattrill** [comment]
Yes.....Drosophila has a heart (I've argued it before and the GO cardiac project decided that too)...

**Comment by ukemi** [comment]
I know. I was part of the GO cardiac project, and the kidney one too!

**Comment by ukemi** [comment]
If flies didn't have a heart, why would they have a mutant called tinman without one? One of my favorites!

**Comment by hattrill** [comment]
If you've ever had to endure a fly mating paper (with or without heads), you'll know that they never short change you on the love!

**Comment by pgaudet** [comment]
Could there be strange behavior when we have to many 'never in' any 'only in'? 

There are issues with taxon checks - we know that some taxon constraints are missed, see https://github.com/geneontology/go-site/issues/1253 and also I have reported similar false positive problems. One issue was when a taxon was not known by owltools (https://github.com/geneontology/go-site/issues/734#issuecomment-413282789) - as far as I know owltools is not running the taxon checks anymore, but there are taxa we dont know of. 

Another similar example to what is reported here:  'GO:0060708 spongiotrophoblast differentiation' also seems to be a false positive report for the human gene O35718 (see http://release.geneontology.org/2020-01-01/reports/goa_human-report.html) 

@kltm @dougli1sqrd @balhoff 

Thanks, Pascale

**Comment by pgaudet** [comment]
@hattrill Do you still have problems with this ?

**Comment by hattrill** [comment]
Yes - still have taxon violations for everything under heart development

**Comment by pgaudet** [comment]
I thought the heart issues were solved with the merge of cardiovascular system development  into 'circulatory system development #19452

@ukemi @balhoff Any idea why this is still failing ?

**Comment by ukemi** [comment]
I just looked at the taxon constraints using @balhoff 's plug-in and it all looks fine to me, both heart development and circulatory system development. Should only be in bilateria, which includes flies. Could it be an update issue? Although I did the merge in July.

**Comment by pgaudet** [comment]
Looking at this with @balhoff 
This is fixed for GO:0003007 heart morphogenesis, but other terms' LD had not been edited, so it still fails. 

The same fix can be made for 'heart formation', but I am not clear what to do about 'heart tube', 'cardiac muscle' ,etc. 

I think we need to go into Uberon and fix taxon constraints and 'parts of'. The problem is that 'heart' will end up disconnected from many other structures. @cmungall is that OK with you ? 

Alternatively we can remove LD. 

Thanks, Pascale

**Comment by cmungall** [comment]
Can this ticket be closed? Do we still get incoherencies from these?

GO has been fixed so it is not using the uberon vertebrate specific structure

id: GO:0007507
name: heart development
namespace: biological_process
alt_id: GO:0007511
def: "The process whose specific outcome is the progression of the heart over time, from its formation to the mature structure. The heart is a hollow, muscular organ, which, by contracting rhythmically, keeps up the circulation of the blood." [GOC:jid, UBERON:0000948]
synonym: "cardiac development" RELATED []
synonym: "dorsal vessel development" NARROW []
xref: Wikipedia:Heart_development
is_a: GO:0032502 ! developmental process
is_a: GO:0048513 ! animal organ development
intersection_of: GO:0048856 ! anatomical structure development
intersection_of: results_in_development_of UBERON:0007100 ! primary circulatory organ
relationship: part_of GO:0072359 ! circulatory system development

id: GO:0003007
name: heart morphogenesis
namespace: biological_process
def: "The developmental process in which the heart is generated and organized. The heart is a hollow, muscular organ, which, by contracting rhythmically, keeps up the circulation of the blood." [GOC:dph, GOC:isa_complete]
synonym: "cardiac morphogenesis" RELATED []
is_a: GO:0009887 ! animal organ morphogenesis
intersection_of: GO:0009653 ! anatomical structure morphogenesis
intersection_of: results_in_morphogenesis_of UBERON:0007100 ! primary circulatory organ
relationship: part_of GO:0007507 ! heart development

Note that robot now exports markdown in explanations for incoherencies, so I suggest we add these to these tickets

**Comment by cmungall** [comment]
it seems the view of this group is that Uberon should relabel "UBERON:0007100 ! primary circulatory organ" to be 'heart', and relabel "UBERON:0000948 ! heart" to be "vertebrate heart", if so someone can make a ticket. But to be clear, this is purely a terminological alignment issue, there never was a dependency on uberon making logical changes to satisfy this ticket

**Comment by pgaudet** [comment]
@hattrill Is this still an issue?

**Comment by hattrill** [comment]
Hi Pascale,
Appears not to be. I will have a look over the taxon issues next week and get back to you on this. Thanks
